### PR TITLE
Makefile modularization

### PR DESCRIPTION
**Pull request by nopeslide**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,120 +1,65 @@
...
+#clean-flash:
+#	@true
 
-.PHONY: distclean-flash
-distclean-flash: clean-flash
-	@true
+#.PHONY: distclean-flash
+#DISTCLEAN += clean-flash
+#distclean-flash: clean-flash
+#	@true
```
Comment (ikstream) [inline Makefile]: No comments, keep it in or remove it

File: Makefile [hunk 2]
```diff
@@ -125,24 +70,25 @@ monitor: | check-monitor
...
+#.PHONY: clean-monitor
+#clean-monitor:
+#	@true
 
-.PHONY: distclean-monitor
-distclean-monitor: clean-monitor
-	@true
+#.PHONY: distclean-monitor
+#distclean-monitor: clean-monitor
+#	@true
```
Comment (ikstream) [inline Makefile]: No comments, keep it in or remove it

File: Makefile [hunk 3]
```diff
@@ -177,19 +124,23 @@ check-dev:
...
-.PHONY: clean-dev
 clean-dev:
 	rm -f .dev
 
-.PHONY: distclean-dev
-distclean-dev: clean-dev
+#.PHONY: distclean-dev
+#distclean-dev: clean-dev
+#	@true
+
```
Comment (ikstream) [inline Makefile]: No comments, keep it in or remove it

**Review by ikstream — CHANGES_REQUESTED** [review]
What's the advantage of moving staff to a hidden file instead of keeping it in just one?

**Comment by nopeslide** [comment]
> What's the advantage of moving staff to a hidden file instead of keeping it in just one?

The same logic exists in `/Makefile` & `/docker/Makefile`, the next step would be to simplify the docker Makefile. I dot-prefixed it so it doesn't clutter the root.

**Review by ikstream — CHANGES_REQUESTED** [review]
Please add some instructions on how to use the updated Makefile

**Review by ikstream — APPROVED** [review]

**Pull request merged**